### ISSUE TITLE
77

### ISSUE THREAD

**Issue by SteveHong1901** [SteveHong1901/STAT0023-Coursework-1#11]
def bootstrap_standard_errors(model, X, y, num_samples=1000):
    """
    Estimate the standard errors of logistic regression coefficients using bootstrapping.
    
    Parameters:
    - model: An instance of the MulticlassLogisticRegression class.
    - X: The input features.
    - y: The target values.
    - num_samples: Number of bootstrap samples to generate.

    Returns:
    - Standard errors for each coefficient.
    """

    # Get the number of data points
    m, n = X.shape
    k = len(np.unique(y))  # Number of classes

    # Store all bootstrap coefficient estimates here
    all_weights = np.zeros((num_samples, n + 1, k))

    for i in range(num_samples):
        # Sample with replacement from X and y
        indices = np.random.choice(m, m, replace=True)
        X_sample = X[indices]
        y_sample = y[indices]

        # Fit the model on the bootstrap sample
        bootstrap_model = MulticlassLogisticRegression(learning_rate=model.learning_rate, num_iterations=model.num_iterations)
        bootstrap_model.fit(X_sample, y_sample)

        # Store the coefficient estimates
        all_weights[i] = bootstrap_model.weights

    # Compute the standard error for each coefficient
    standard_errors = all_weights.std(axis=0)

    return standard_errors

# Usage:
# model = MulticlassLogisticRegression()
# model.fit(X_train, y_train)
# std_errors = bootstrap_standard_errors(model, X_train, y_train)
# print(std_errors)
